doc/manual/src/contributing/experimental-features.md

```diff
@@ -0,0 +1,64 @@
+This section describes the notion of *experimental features*, and how it fits into the big picture of the development of Nix.
+
+# What are experimental features?
+
+[Since Nix 2.4](../release-notes/rl-2.4.md), some features of Nix can be flagged as experimental.
+Experimental features can be changed or removed at any time, and users must [explicitly enable them](../nix-conf.md#conf-experimental-features).
+This allows accessing new functionality that is not yet stable without unwittingly relying on them.
+
+# When should a new feature be marked experimental?
+
+A change in the Nix codebase should be guarded by an experimental feature flag if it is considered likely to be reverted or adapted in a backwards-incompatible manner after gathering more experience with it in practice.
+
+Examples:
```

Comment:
```suggestion
Examples of larger and highly visible requiring experimental features:
```

Reply:
(It was fine before, but if we add my longer description above, I think it stops becoming clear what this is an example of :)